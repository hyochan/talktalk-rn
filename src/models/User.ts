import { ImageURISource } from 'react-native';
import { Chatroom } from './Chatroom';

export class User {
  public constructor(
    public uid: string = '',
    public displayName: string = '',
    public photoURL: string = '',
    public statusMsg: string = '',
    public isOnline: boolean = false,
    public friends: User[] = [],
    public chatrooms: Chatroom[] = [],
    public created: Date = undefined,
    public updated: Date = undefined,
  ) {}
}
