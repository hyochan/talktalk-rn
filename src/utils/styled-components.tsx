import * as styledComponents from "styled-components/native";
import ThemeInterface from "./theme";

const {
  default: styled,
  css,
  ThemeProvider
} = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<typeof ThemeInterface>;

export { css, ThemeProvider };
export default styled;