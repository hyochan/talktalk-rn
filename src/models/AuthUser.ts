import { User } from './User';

export class AuthUser extends User {
  constructor(private pushToken: string = '') {
    super();
  }
}
