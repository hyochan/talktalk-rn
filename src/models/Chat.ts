import { User } from './User';

export class Chat {
  public constructor(
    public id: string = '', 
    public sender: User = undefined, 
    public message: string = '', 
    public created: Date = undefined, 
    public updated: Date = undefined
  ) {}
}
