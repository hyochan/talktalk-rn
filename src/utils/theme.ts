import { Dimensions, Platform } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import { DefaultTheme } from 'styled-components';

const { width, height } = Dimensions.get('window');
let calRatio = width <= height ? 16 * (width / height) : 16 * (height / width);
if (width <= height) {
  if (calRatio < 9) {
    calRatio = width / 9;
  } else {
    calRatio = height / 18;
  }
} else {
  if (calRatio < 9) {
    calRatio = height / 9;
  } else {
    calRatio = width / 18;
  }
}

const theme: DefaultTheme = {
  screenWidth: width,
  screenHeight: height,
  ratio: calRatio / (360 / 9),
  statusBarHeight: getStatusBarHeight(true),
  colors: {
    background: '#e3e3e3',
    dodgerBlue: 'rgb(58,139,255)',
    dusk: 'rgb(65,77,107)',
    blueyGray: 'rgb(134,154,183)',
    cloudyBlue: 'rgb(175,194,219)',
    greenishCyan: 'rgb(80,227,194)',
    paleGray: 'rgb(233,237,244)',
  }
};

export default theme;