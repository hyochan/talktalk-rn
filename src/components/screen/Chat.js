// @flow
import React, {
  Component,
  useState,
  useEffect,
  useRef,
} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  Keyboard,
  Platform,
  TextInput,
  KeyboardAvoidingView,
  FlatList,
} from 'react-native';
import styled from 'styled-components/native';

import { Header } from 'react-navigation';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import ChatListItem from '../shared/ChatListItem';
import EmptyListItem from '../shared/EmptyListItem';
import Button from '../shared/Button';

import { Chatroom } from '../../models/Chatroom';
import { IC_SMILE } from '../../utils/Icons';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';
import { Chat } from '../../models/Chat';
import { User } from '../../models/User';

const StyledContainer = styled.SafeAreaView`
  flex: 1;
  background-color: white;
  flex-direction: column;
  align-items: center;
`;

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
  justify-content: center;
  align-self: stretch;

  flex-direction: column;
  align-items: center;
`;

const StyledViewChat = styled.View`
  width: 100%;
  border-top-width: 0.5;
  border-color: rgb(225,225,225);
  background-color: white;
  min-height: 52;
  max-height: 100;

  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
`;

const StyledInputChat = styled.TextInput`
  width: 80%;
  font-size: 14;
  margin-right: 20;
  padding-left: 48;
  color: black;
`;

const StyledTouchMenu = styled.TouchableOpacity`
  position: absolute;
  left: 10;
  height: 100%;
  minWidth: 20;
  justifyContent: center;
`;

const StyledViewBottom = styled.View`
  position: absolute;
  bottom: 0;
  width: 100%;
`;

const StyledViewMenu = styled.View`
  height: 258;
  background-color: green;
`;

type Styles = {
  btnSend: ViewStyle,
  txtSend: TextStyle,
};

const styles: Styles = StyleSheet.create({
  btnSend: {
    right: 8,
    backgroundColor: colors.dodgerBlue,
    borderRadius: 4,
    width: 60,
    height: 36,

    alignItems: 'center',
    justifyContent: 'center',
  },
  txtSend: {
    fontSize: 14,
    fontWeight: 'bold',
    color: 'white',
    paddingHorizontal: 5,
    paddingVertical: 10,
  },
});

type Props = {
  navigation: any;
};
type State = {
  isLoading: boolean,
  showMenu: boolean,
  message: string,
  chats: Chat[],
};

function Screen(props: Props, state: State) {
  let keyboardDidShowListener: any;
  let keyboardDidHideListener: any;

  const input1 = useRef(null);
  const input2 = useRef(null);
  const [isLoading, setIsLoading] = useState(false);
  const [showMenu, setShowMenu] = useState(false);
  const [message, setMessage] = useState('');
  const [chats, setChats] = useState([
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'hello1',
    ),
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'Hello2. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message.',
    ),
    new Chat(
      '',
      new User(
        '1',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message.',
    ),
  ]);

  useEffect(() => {
    if (showMenu) {
      Keyboard.dismiss();
    } else {
      if (input1 && input1.current) {
        input1.current.focus();
      }
    }
  }, [showMenu]);

  useEffect(() => {
    Keyboard.addListener('keyboardDidShow', (e) => {
      console.log('keyboardHeight', e.endCoordinates.height);
    });
    return () => {
      if (keyboardDidShowListener) {
        keyboardDidShowListener.remove();
      }
    };
  });

  const renderItem = ({ item, index } : any) => {
    return (
      <ChatListItem
        prevItem={index > 0 ? chats[ index - 1 ] : null}
        item={item}
      />
    );
  };

  const sendChat = () => {
    console.log('sendChat');
  };

  const goBack = () => {
    props.navigation.goBack();
  };

  return (
    <StyledContainer>
      <StyledKeyboardAvoidingView
        keyboardVerticalOffset = {Header.HEIGHT + 24}
        behavior={Platform.select({
          'android': null,
          'ios': 'padding',
        })}
      >
        <FlatList
          style={{ alignSelf: 'stretch' }}
          contentContainerStyle={
            chats.length === 0
              ? {
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center',
              }
              : null
          }
          keyExtractor={(item, index) => index.toString()}
          data={chats}
          renderItem={renderItem}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
        {
          !showMenu
            ? <StyledViewChat>
              <StyledInputChat
                ref={input1}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={ colors.cloudyBlue }
                value={message}
                defaultValue={message}
                onChangeText={(text) => setMessage(text)}
              />
              <StyledTouchMenu
                onPress={() => setShowMenu(true)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                isLoading={isLoading}
                onPress={sendChat}
                style={styles.btnSend}
                textStyle={styles.txtSend}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            : null
        }
      </StyledKeyboardAvoidingView>
      {
        showMenu
          ? <StyledViewBottom>
            <StyledViewChat>
              <StyledInputChat
                ref={input2}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={ colors.cloudyBlue }
                value={message}
                defaultValue={message}
              />
              <StyledTouchMenu
                onPress={() => setShowMenu(false)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                isLoading={isLoading}
                onPress={sendChat}
                style={styles.btnSend}
                textStyle={styles.txtSend}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            <StyledViewMenu />
          </StyledViewBottom>
          : null
      }
    </StyledContainer>
  );
}

export default Screen;
