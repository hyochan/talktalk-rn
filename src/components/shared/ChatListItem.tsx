import React, { Component } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  Image,
  Text,
  View,
  Dimensions,
  PixelRatio,
  TouchableOpacity,
} from 'react-native';

import styled from 'styled-components/native';
import LinearGradient from 'react-native-linear-gradient';
import moment from 'moment';
import Icon5 from 'react-native-vector-icons/FontAwesome5';

import { Chat } from '../../models/Chat';
import { ratio, colors, screenWidth } from '../../utils/Styles';
import { User } from '../../models/User';

const StyledWrapperPeer = styled.View`
  min-height: 48;
  flexDirection: row;
  alignItems: flex-start;
  justifyContent: flex-start;
  margin-left: 20;
  margin-right: 8;
`;

const StyledImageSender = styled.Image`
  width: 32;
  height: 32;
`;

const StyledTextPeerMessage = styled.Text`
  font-size: 14;
  color: ${colors.dusk};
`;

const StyledTextPeerName = styled.Text`
  font-size: 12;
  color: ${colors.dusk};
  margin-bottom: 2;
`;

const StyledTextPeerDate = styled.Text`
  font-size: 12;
  color: ${colors.cloudyBlue};
  margin-right: 20;
`;

const StyledWrapperMy = styled.View`
  min-height: 48;
  width: 100%;
  margin-top: 20;

  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-end;
`;

const StyledTextDate = styled.Text`
  font-size: 12;
  color: ${colors.cloudyBlue},
  margin-top: 4;
  margin-right: 20;
`;

const StyledTextMessage = styled.Text`
  font-size: 14;
  color: white;
`;

interface IProps {
  item: Chat;
  prevItem?: Chat;
  onPressPeerImage?: VoidFunction;
}

const myFakeUid = '2'; // TODO: temporary

function Shared(props: IProps) {
  const isSamePeerMsg = props.prevItem && props.prevItem.sender.uid === props.item.sender.uid;
  return (
    props.item.sender.uid !== myFakeUid // peer message
      ? <StyledWrapperPeer style={{
        marginTop: isSamePeerMsg ? 0 : 20,
        marginRight: 80,
      }}>
        <View style={{ marginRight: 8 }}>
          <TouchableOpacity
            testID='peer_image'
            onPress={props.onPressPeerImage}
          >
            {
              props.item.sender.photoURL !== ''
                ? <StyledImageSender source={props.item.sender.photoURL}/>
                : isSamePeerMsg
                  ? <View style={{ width: 40 }} />
                  : <Icon5 name='meh' size={40} color={colors.dusk} light/>
            }
          </TouchableOpacity>
        </View>
        <View style={{ flexDirection: 'column' }}>
          {
            isSamePeerMsg
              ? <View/>
              : <StyledTextPeerName>{props.item.sender.displayName}</StyledTextPeerName>
          }
          <View style={{
            marginRight: 8,
            backgroundColor: 'white',
            borderRadius: 3,
            borderWidth: 1,
            borderColor: 'rgb(225,225,225)',
            padding: 12,
            shadowColor: colors.paleGray,
            shadowOffset: {
              width: 0,
              height: 2,
            },
            marginVertical: 4,
          }}><StyledTextPeerMessage
            >{props.item.message}</StyledTextPeerMessage>
          </View>
          <StyledTextPeerDate>
            {
              props.item.created
                ? `${moment(props.item.created).hour()} : ${moment(props.item.created).minutes()}`
                : '0 : 0'
            }
          </StyledTextPeerDate>
        </View>
      </StyledWrapperPeer>
      : <StyledWrapperMy>
        <LinearGradient
          start={{ x: 0.2, y: 0.4 }} end={{ x: 1.0, y: 0.8 }}
          locations={[0, 0.85]}
          colors={['rgb(100,152,212)', colors.dodgerBlue]}
          style={{
            marginHorizontal: 20,
            marginLeft: 8,
            padding: 12,
            borderRadius: 3,
          }}>
          <StyledTextMessage>{props.item.message}</StyledTextMessage>
        </LinearGradient>
        <StyledTextDate>
          {
            props.item.created
              ? `${moment(props.item.created).hour()} : ${moment(props.item.created).minutes()}`
              : '0 : 0'
          }
        </StyledTextDate>
      </StyledWrapperMy>
  );
}

Shared.defaultProps = {
  item: new Chat(
    '',
    new User('', '', '', ''),
    '',
  ),
};

export default Shared;
