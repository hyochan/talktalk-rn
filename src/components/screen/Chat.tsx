import React, {
  Component,
  useState,
  useEffect,
  useRef,
} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  Keyboard,
  Platform,
  TextInput,
  KeyboardAvoidingView,
  FlatList,
} from 'react-native';
import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';
import styled, { ThemeProps, withTheme, DefaultTheme } from 'styled-components/native';

import { Header } from 'react-navigation';
import Icon5, { FA5Style } from 'react-native-vector-icons/FontAwesome5';
import ChatListItem from '../shared/ChatListItem';
import EmptyListItem from '../shared/EmptyListItem';
import Button from '../shared/Button';

import { Chatroom } from '../../models/Chatroom';
import { IC_SMILE } from '../../utils/Icons';
import { getString } from '../../../STRINGS';
import { Chat } from '../../models/Chat';
import { User } from '../../models/User';

const StyledContainer = styled.SafeAreaView`
  flex: 1;
  background-color: white;
  flex-direction: column;
  align-items: center;
`;

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
  justify-content: center;
  align-self: stretch;
  flex-direction: column;
  align-items: center;
`;

const StyledViewChat = styled.View`
  width: 100%;
  border-top-width: 0.5;
  border-color: rgb(225,225,225);
  background-color: white;
  min-height: 52;
  max-height: 100;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
`;

const StyledInputChat = styled.TextInput`
  width: 80%;
  font-size: 14;
  margin-right: 20;
  padding-left: 48;
  color: black;
`;

const StyledTouchMenu = styled.TouchableOpacity`
  position: absolute;
  left: 10;
  height: 100%;
  min-width: 20px;
  justify-content: center;
`;

const StyledViewBottom = styled.View`
  position: absolute;
  bottom: 0;
  width: 100%;
`;

const StyledViewMenu = styled.View<{ height: number }>`
  background-color: ${({ theme: { colors: { paleGray }}}) => paleGray};
  flex-direction: row;
  flex-wrap: wrap;
  height: ${({ height }) => height};
`;

interface IProps extends ThemeProps<DefaultTheme> {
  navigation: any;
}

interface IState {
  isLoading: boolean;
  showMenu: boolean;
  message: string;
  chats: Chat[];
}

function Screen(props: IProps, state: IState) {
  let keyboardShowListener: any;

  const input1 = useRef(null);
  const input2 = useRef(null);
  const [keyboardHeight, setKeyboardHeight] = useState(258);
  const [isLoading, setIsLoading] = useState(false);
  const [showMenu, setShowMenu] = useState(false);
  const [message, setMessage] = useState('');
  const [chats, setChats] = useState([
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'hello1',
    ),
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'Hello2. This is long message. This is long message. This is long message.This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message.',
    ),
    new Chat(
      '',
      new User(
        '1',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message.',
    ),
  ]);

  useEffect(() => {
    if (showMenu) {
      Keyboard.dismiss();
    } else {
      if (input1 && input1.current) {
        input1.current.focus();
      }
    }
  }, [showMenu]);

  useEffect(() => {
    keyboardShowListener = Keyboard.addListener('keyboardDidShow', (e) => {
      setKeyboardHeight(e.endCoordinates.height);
    });
    return () => {
      if (keyboardShowListener) {
        keyboardShowListener.remove();
      }
    };
  });

  const renderItem = ({ item, index }: any) => {
    return (
      <ChatListItem
        prevItem={index > 0 ? chats[ index - 1 ] : null}
        item={item}
      />
    );
  };

  const sendChat = () => {
  };

  const {
    theme: {
      colors: {
        dusk,
        cloudyBlue
      }
    }    
  } = props;

  return (
    <StyledContainer>
      <StyledKeyboardAvoidingView
        keyboardVerticalOffset={Header.HEIGHT}
        behavior={Platform.select({
          android: null,
          ios: 'padding',
        })}
      >
        <FlatList
          style={{ alignSelf: 'stretch' }}
          contentContainerStyle={
            chats.length === 0
              ? {
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center',
              }
              : null
          }
          keyExtractor={(item, index) => index.toString()}
          data={chats}
          renderItem={renderItem}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
        {
          !showMenu
            ? <StyledViewChat>
              <StyledInputChat
                testID='input_chat'
                ref={input1}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={cloudyBlue}
                value={message}
                defaultValue={message}
                onChangeText={(text) => setMessage(text)}
              />
              <StyledTouchMenu
                testID='touch_menu'
                onPress={() => setShowMenu(true)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                testID='btn_chat'
                isLoading={isLoading}
                onPress={sendChat}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            : null
        }
      </StyledKeyboardAvoidingView>
      {
        showMenu
          ? <StyledViewBottom>
            <StyledViewChat>
              <StyledInputChat
                ref={input2}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={cloudyBlue}
                value={message}
                defaultValue={message}
              />
              <StyledTouchMenu
                testID='touch_menu'
                onPress={() => setShowMenu(false)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                testID='btn_chat'
                isLoading={isLoading}
                onPress={sendChat}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            <StyledViewMenu testID="viewMenu" height={keyboardHeight}>
              <View style={{
                flexDirection: 'row',
                marginTop: 10,
              }}>
                <TouchableOpacity
                  style={{
                    marginLeft: 16,
                    marginTop: 2,
                    width: 60,
                    height: 60,
                    justifyContent: 'center',
                    alignItems: 'center',
                  }}
                >
                  <Icon5 name='camera' size={36} color={dusk} light />
                </TouchableOpacity>
                <TouchableOpacity
                  style={{
                    marginLeft: 16,
                    marginTop: 4,
                    width: 60,
                    height: 60,
                    justifyContent: 'center',
                    alignItems: 'center',
                  }}
                >
                  <Icon5 name='image' size={40} color={dusk} light/>
                </TouchableOpacity>
              </View>
            </StyledViewMenu>
          </StyledViewBottom>
          : null
      }
    </StyledContainer>
  );
}

export default withTheme(Screen);
