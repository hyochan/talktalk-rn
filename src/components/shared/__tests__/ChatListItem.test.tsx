import 'react-native';
import * as React from 'react';
import ChatListItem from '../ChatListItem';
import { ThemeProvider } from 'styled-components/native';
import theme from '../../../utils/theme';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import { Chat } from '../../../models/Chat';
import { User } from '../../../models/User';

let cnt = 0;
const onPressPeerImage = () => {
  cnt++;
};

const props = {
  item: new Chat(
    '',
    new User('', '', '', ''),
    '',
  ),
  prevItem: new Chat(
    '',
    new User('', '', '', ''),
    '',
  ),
  onPressPeerImage,
  theme
}
const component: React.ReactElement = (
  <ThemeProvider theme={theme}>
    <ChatListItem {...props} />
  </ThemeProvider> 
);

describe('[ChatListItem] rendering test', () => {
  it('renders [peerMessage] as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[ChatListItem] interaction', () => {
  let rendered: renderer.ReactTestRenderer;
  let testingLib: any;

  beforeAll(() => {
    rendered = renderer.create(component);
    testingLib = render(component);
  });

  it('should fireEvent when peer image is pressed', () => {
    const touchPeerImage = testingLib.getByTestId('peer_image');
    fireEvent(touchPeerImage, 'press');
    expect(cnt).toEqual(1);
  });
});
