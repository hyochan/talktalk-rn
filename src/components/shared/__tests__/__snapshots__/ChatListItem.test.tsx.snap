// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ChatListItem] rendering test renders [peerMessage] as expected 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "marginLeft": 20,
        "marginRight": 8,
        "minHeight": 48,
      },
      Object {
        "marginRight": 80,
        "marginTop": 20,
      },
    ]
  }
>
  <View
    style={
      Object {
        "marginRight": 8,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="peer_image"
    >
      <Unknown
        color="rgb(65,77,107)"
        light={true}
        name="meh"
        size={40}
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgb(65,77,107)",
            "fontSize": 12,
            "marginBottom": 2,
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "rgb(225,225,225)",
          "borderRadius": 3,
          "borderWidth": 1,
          "marginRight": 8,
          "marginVertical": 4,
          "padding": 12,
          "shadowColor": "rgb(233,237,244)",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgb(65,77,107)",
              "fontSize": 14,
            },
          ]
        }
      >
        
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "rgb(175,194,219)",
            "fontSize": 12,
            "marginRight": 20,
          },
        ]
      }
    >
      0 : 0
    </Text>
  </View>
</View>
`;
